app/src/org/commcare/activities/connect/ConnectActivity.java

```diff
@@ -14,6 +19,17 @@
 
 public class ConnectActivity extends CommCareActivity<ResourceEngineListener> {
 
+    ActivityResultLauncher<Intent> verificationLauncher = registerForActivityResult(
+            new ActivityResultContracts.StartActivityForResult(),
+            result -> {
+                if (result.getResultCode() == Activity.RESULT_OK) {
+                    ConnectDownloadingFragment connectDownloadFragment = getConnectDownloadFragment();
+                    if (connectDownloadFragment != null) {
+                        connectDownloadFragment.startLearning();
```

Comment:
This appears to be hard-coded for the learning app, but we'll need the functionality for both learn and delivery apps. There's an "isLearning" boolean now being passed into ConnectDownloadingFragment, maybe just a matter of accessing it here to either call startLearning or startDelivery?

Reply:
`startLearning` does do the learn or deliver correctly but I have [renamed the method ](https://github.com/dimagi/commcare-android/pull/2719/commits/5f405dbde9b3e3832b9e6cfdedab138c1f6e1b70) to avoid the confusion

Reply:
Ah, that makes sense but thanks for making it more clear.